metaseq/checkpoint_utils.py

```diff
@@ -105,77 +84,34 @@ def is_better(a, b):
     checkpoints = [
         os.path.join(cfg.save_dir, fn) for fn, cond in checkpoint_conds.items() if cond
     ]
-    if len(checkpoints) > 0:
-        if PathManager.islink(checkpoints[0]):
-            PathManager.rm(checkpoints[0])
-
-        trainer.save_checkpoint(
-            checkpoints[0],
-            extra_state,
-            training_finished=training_finished,
-            async_callback_fn=async_callback_fn,
-        )
-
-        def _copy_if_not_async(src, dest):
-            if cfg.write_checkpoints_asynchronously:
-                pass  # TODO[ioPath]: Need to implement a delayed asynchronous file copying/moving feature.
-            else:
-                assert PathManager.copy(
-                    src, dest, overwrite=True
-                ), f"Failed to copy {src} to {dest}"
 
-        for cp in checkpoints[1:]:
-            _copy_if_not_async(src=checkpoints[0], dest=cp)
+    if PathManager.islink(checkpoints[0]):
+        PathManager.rm(checkpoints[0])
 
-        write_timer.stop()
-        logger.info(
-            "Saved checkpoint {} (epoch {} @ {} updates, score {}) (writing took {} seconds)".format(
-                checkpoints[0], epoch, updates, val_loss, write_timer.sum
-            )
-        )
-
-    _delete_old_checkpoint_files(
```

Comment:
cc @aitorormazabal given your latest PR: https://github.com/facebookresearch/metaseq/pull/341 - it seems like you are calling into this codepath; what kind of checkpoint deletion frequency/policy are you looking for?

Reply:
Yes I am using `--keep-interval-updates 2 ` to prevent my checkpoint dirs from growing too much. I find the `keep_` functionality quite useful for running experiments with limited disk space

Reply:
Thanks @aitorormazabal - sorry @Xirider we're now using your PR to gather requirements live 😅 . Seems like we should include a "keep last N checkpoints" logic to handle the above. 

Net diff in this PR will now give us only two knobs: checkpoint every N number of iterations, keep last K checkpoints (note that each checkpoint can be many files, so this "last K" logic should account for that; would be good to confirm since we haven't turned this on for large-scale training yet).

Reply:
😂 @suchenzang So do we want still want to be able to save after epochs? And do we want to save after a run is finished early?

--save-interval
--save-interval-updates

--keep-interval-updates
--keep-interval

or just:

--save-interval-updates
--keep-interval-updates

Reply:
Ah yeah forgot we're keeping those two (epoch vs interval) separate 🤦‍♀️ . So yes, the first case:
```
--save-interval
--save-interval-updates

--keep-interval
--keep-interval-updates
```
where the former is a frequency, and the latter is a count (hopefully that isn't confusing to distinguish between).

To walk through an example: I would imagine wanting to set save every epoch on + save every 100 steps, followed by keep last epoch (evict older ones) + last 5 checkpoints (mapping to 500 steps) so that we don't ever have more than 1+5 checkpoints around on disk (in the above example).

Reply:
Yes, that makes sense. Thanks for the example!